vllm/entrypoints/llm.py

```diff
@@ -1042,40 +1039,87 @@ def ensure_str(prompt: SingletonPrompt):
-            parsed_prompts.append(engine_prompt)
+            tokenization_kwargs: Dict[str, Any] = {}
+            if truncate_prompt_tokens is not None:
+                tokenization_kwargs["truncation"] = True
+                tokenization_kwargs["max_length"] = truncate_prompt_tokens
+
+            parsed_prompts = []
+
+            for q, t in input_pairs:
+                prompt_inputs = tokenizer(text=q,
+                                          text_pair=t,
+                                          **tokenization_kwargs)
+                engine_prompt = TokensPrompt(
+                    prompt_token_ids=prompt_inputs["input_ids"],
+                    token_type_ids=prompt_inputs.get("token_type_ids"))
+                parsed_prompts.append(engine_prompt)
+
+            self._validate_and_add_requests(
+                prompts=parsed_prompts,
+                params=pooling_params,
+                lora_request=lora_request,
+                prompt_adapter_request=prompt_adapter_request,
+            )
 
-        self._validate_and_add_requests(
-            prompts=parsed_prompts,
-            params=pooling_params,
-            lora_request=lora_request,
-            prompt_adapter_request=prompt_adapter_request,
-        )
+            outputs = self._run_engine(use_tqdm=use_tqdm)
+            items = self.engine_class.validate_outputs(outputs,
+                                                       PoolingRequestOutput)
 
-        outputs = self._run_engine(use_tqdm=use_tqdm)
-        items = self.engine_class.validate_outputs(outputs,
-                                                   PoolingRequestOutput)
+            return [ScoringRequestOutput.from_base(item) for item in items]
+
+        elif self.llm_engine.model_config.runner_type == "pooling":
+            text = text_1 + text_2
+            encoded_text = self.encode(text)
+            encoded_text_1 = encoded_text[0:len(text_1)]
+            encoded_text_2 = encoded_text[len(text_1):]
+
+            if len(encoded_text_1) == 1:
+                encoded_text_1 = encoded_text_1 * len(encoded_text_2)
+
+            output_pairs = [(t1, t2)
+                            for t1, t2 in zip(encoded_text_1, encoded_text_2)]
+
+            scores = []
+            cosSim = torch.nn.CosineSimilarity(0)
+
+            for embed_1, embed_2 in output_pairs:
+                pairs_score = cosSim(embed_1.outputs.data,
+                                     embed_2.outputs.data)
+
+                if getattr(tokenizer, "pad_token", None) is None:
+                    tokens = embed_1.prompt_token_ids + embed_2.prompt_token_ids
+                else:
+                    tokens = embed_1.prompt_token_ids + [
+                        tokenizer.pad_token_type_id
+                    ] + embed_2.prompt_token_ids
+
+                scores.append(
+                    PoolingRequestOutput(
+                        request_id="unk",  #unk ou id incremental?
```

Comment:
Some other ideas:

- Use the first request ID from the batch: `encoded_text[0].request_id`
- Use the first and last request ID from the batch `f"{encoded_text[0].request_id}_{encoded_text[-1].request_id}"` 
- Go super verbose and put all of them: `"_".join(x.request_id for x in encoded_text)`

Reply:
Since we have the IDs for all embedding results, we could grab the ids and concatenate them for each scoring output:
```python
                pair_id  = f"{embed_1.request_id}_{embed_2.request_id}"
```